apps/google-analytics-4/frontend/src/helpers/signed-requests.ts

```diff
@@ -0,0 +1,38 @@
+import { CreateAppSignedRequestProps } from 'contentful-management/dist/typings/entities/app-signed-request';
+import { PlainClientAPI } from 'contentful-management/dist/typings/plain/common-types';
+
+async function makeSignedRequest(
+  url: URL,
+  appDefinitionId: string,
+  cma: PlainClientAPI,
+  method: CreateAppSignedRequestProps['method'] = 'GET',
+  headers: {} = {} // TOOD: improve HTTP request headers typing
+): Promise<Response> {
+  const req = {
+    url: url,
+    method: method,
+    headers: headers,
+  };
+
+  // add request verification signing secret to request headers
+  const { additionalHeaders } = await cma.appSignedRequest.create(
+    {
+      appDefinitionId: appDefinitionId,
+    },
+    {
+      method: req.method,
+      headers: req.headers,
+      path: new URL(req.url).pathname,
+    }
+  );
+  Object.assign(req.headers, additionalHeaders);
+
+  // fetch google analytics credentials for the current entry
+  return fetch(req.url, req).then((res) => {
+    if (res.status === 403) throw new Error('Unauthorized');
```

Comment:
This might be good for testing but I recommend we either:

* Just return the request without any handling. The 403 can get handled by the user of this function.
* Raise a more specific error that allows us to intentionally handle failed signature matches. Right now we have more than one behavior in our backend that oculd return a 403, e.g. you're not authorized to hit a Google API. This logic would swallow those errors. To avoid I think we should look at the message body (for an errorType indicating failed signature) and then raise a custom error, which can get handled.

Reply:
refactored to not parse the response or throw error.